### ISSUE TITLE
False positive for rule 921110

### ISSUE THREAD

**Issue by jdevreese** [coreruleset/coreruleset#2054]
### Description

Uploading an xml containing text `budget foo)</bar>\n` triggers this false positive.

### Audit Logs / Triggered Rule Numbers

Rule ID: 921110

> ```[13/Apr/2021:14:41:44 +0200] 161831770463.585568 172.19.0.167 63907 192.168.1.151 443 ModSecurity: Warning. Matched "Operator `Rx' with parameter `(?:get|post|head|options|connect|put|delete|trace|track|patch|propfind|propatch|mkcol|copy|move|lock|unlock)\s+(?:\/|\w)[^\s]*(?:\s+http\/\d|[\r\n])' against variable `REQUEST_BODY' (Value: `------WebKitFormBoundaryFYfRC2r9AA3TKtgx\x0d\x0aContent-Disposition: form-data; name="file"; filenam (303055 characters omitted)' ) [file "/etc/modsecurity.d/owasp-crs/rules/REQUEST-921-PROTOCOL-ATTACK.conf"] [line "33"] [id "921110"] [rev ""] [msg "HTTP Request Smuggling Attack"] [data "Matched Data: get foo)</bar>\x0a found within REQUEST_BODY: ------webkitformboundaryfyfrc2r9aa3tktgx\x0d\x0acontent-disposition: form-data; name="file"; filename="redacted (278202 characters omitted)"] [severity "2"] [ver "OWASP_CRS/3.3.0"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/210/272/220/33"] [hostname "192.168.1.151"] [uri "/redacted"] [unique_id "161831770463.585568"] [ref "o67861,34o117443,34v2811,278366t:urlDecodeUni,t:htmlEntityDecode,t:lowercase"]```
### Your Environment

* CRS version: 3.3.0
* ModSecurity version: 3.0.4
* Web Server and version: Nginx 1.19.8

### Confirmation

[X] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

Probably caused by the changes in #1883

**Comment by azurit** [comment]
Thank you for reporting this!

> CRS version: 3.3.0
> Probably caused by the changes in #1883

#1883 is a reported issue so there are no changes caused by it. This issue was resolved by PR https://github.com/coreruleset/coreruleset/pull/1966, are you talking about this PR? If so, can you be more specific in CRS version which you are using? PR https://github.com/coreruleset/coreruleset/pull/1966 was not part of CRS 3.3.0 (release on 1 Jul 2020).

**Comment by azurit** [comment]
@jdevreese Can you, please, provide more info? See above. Thank you!

**Comment by jdevreese** [comment]
Must have made a mistake referencing that issue.

But I was looking at this commit: https://github.com/coreruleset/coreruleset/commit/8f538256366340bb45aff370b058739bf70e04a0

Before: https://regex101.com/r/IgQJtL/2
After: https://regex101.com/r/qSMFIv/2

Suggestion: check for word boundary before the keyword?
https://regex101.com/r/JpmJOC/1

**Comment by azurit** [comment]
@franbuehler Can you, please, explain why have you removed `[\n\r]+` from regexp of rule `921110`? See above. Thank you!

**Comment by franbuehler** [comment]
Ah, I remember this PR. I'll check it.

**Comment by dune73** [comment]
We talked about this in the [April issue chat](https://github.com/coreruleset/coreruleset/issues/2051#issuecomment-813570935). Here is our conclusion:

We lack the context of the commit that @franbuehler explained above. We need to wait for her to provide that context after her holidays. We can then look into the context and ideally come up with a better PR that does not lead to that many new false positives.

**Comment by franbuehler** [comment]
Hi @theMiddleBlue 
You helped me with solving the issue reported by Amit Klein. 

Current regex and coverage is: https://regex101.com/r/WolWP9/1
New proposal would be: https://regex101.com/r/vOG4RW/1

We see that with the word boundary we don't find both HTTP requests.
But to be honest, I don't know if the finding of the second HTTP request is enough. If yes, we can add the word boundary.

Can you help me and do you know if the new proposal would be enough to find the request smuggling??

Thank you :-)

**Comment by theMiddleBlue** [comment]
Hi,

thank you for the recap! I tried the new regex but it seems prone to similar false positives such as `(get it).\n`. Maybe to reduce FP in PL1 we should remove the `|[\r\n]` part from `(?:\s+http\/\d|[\r\n])`. Also, we can move a copy of the actual rule to PL2 to have a stricter rule? What do you think about it?

Thanks!
